WordPress/gutenberg
# Add filter to disable Font Library UI

## What?

This PR adds a filter to disable the Font Library UI at the site level. 

## Why?

Part of the feedback in the introduction of the Font Library was to give developers more control, including disabling it if needed. 

Solves #55275 

## How?
Adds a setting to the editor (`enableFontLibrary`) and conditionally renders the Font Library UI based on this setting.

## Testing Instructions
1. Add the following snippet to a plugin or theme: 
```
function disable_font_library( $editor_settings ){
	$editor_settings['enableFontLibrary'] = false;
	return $editor_settings;
}
add_filter( 'block_editor_settings_all', 'disable_font_library', 10, 1 );
```
2. Go to the site editor > global styles > typography section
3. Verify the Font Library UI is not there

## Screenshots or screencast 
![Screenshot 2024-01-11 at 3 36 13 PM](https://github.com/WordPress/gutenberg/assets/5375500/648d628d-b310-4988-adcd-b70983558a96)
